### PR TITLE
Const for CeedQFunctionGetSourcePath

### PR DESCRIPTION
**Pull request by jeremylt**
Missed a `const`

### CODE REVIEW

**Comment by jedbrown** [comment]
Please hold off on this; I have a few more.

**Comment by jeremylt** [comment]
There are a few that have to be non-const. Some of them are intended for the user to free the string after they are done

**Comment by jeremylt** [comment]
`CeedQFunctionLoadSourceToBuffer` needs to be non-`const`, but `CeedQFunctionGetFields` and `CeedOperatorGetFields` probably should be swapped to make the returned lists const (though that probably gets messy in the Rust interface) as well probably `CeedCompositeOperatorGetSubList` needs to be `const`.

File: interface/ceed-jit-tools.c
Comment (jeremylt) [inline interface/ceed-jit-tools.c:208]: I'm a little confused by adding `&` here, and I think CI doesn't like it

**Comment by jeremylt** [comment]
Ahh, I didn't realize you could `free` a `const` like that. It doesn't always line up with my expectations

File: interface/ceed-jit-tools.c
Comment (jedbrown) [reply interface/ceed-jit-tools.c]: Oh, oops. I had first adjusted things to put the `const` here and use a local variable, then backpedaled and moved it up one frame, but forgot to revert this line.

**Comment by jeremylt** [comment]
Ok, looks like CI is happy. I'm fine merging with the SYCL change since it should be fine and mirrors the changes in CUDA/HIP that pass

**Comment by jedbrown** [comment]
Nope, wait a minute. I reproduced and will fix things on Noether.

**Comment by jedbrown** [comment]
Should be good now once the pipeline passes. To build, I use:

```console
$ . /opt/intel/oneapi/setvars.sh
$ make CC=icx CXX=icpx SYCL_DIR=/opt/intel/oneapi/compiler/2023.2.0/linux
```
We could make a new job in `.gitlab-ci.yml` that only builds.

**Comment by jeremylt** [comment]
Small issue in the CUDA/HIP backends that's fixed now
```
     SYCLCXX build/backends/sycl/ceed-sycl-common.o
     SYCLCXX build/backends/sycl/ceed-sycl-compile.o
     SYCLCXX build/backends/sycl/online_compiler.o
     SYCLCXX build/backends/sycl-ref/ceed-sycl-ref-basis.o
     SYCLCXX build/backends/sycl-ref/ceed-sycl-ref-operator.o
     SYCLCXX build/backends/sycl-ref/ceed-sycl-ref-qfunction-load.o
     SYCLCXX build/backends/sycl-ref/ceed-sycl-ref-qfunction.o
     SYCLCXX build/backends/sycl-ref/ceed-sycl-ref-qfunctioncontext.o
     SYCLCXX build/backends/sycl-ref/ceed-sycl-ref.o
     SYCLCXX build/backends/sycl-ref/ceed-sycl-restriction.o
     SYCLCXX build/backends/sycl-ref/ceed-sycl-vector.o
     SYCLCXX build/backends/sycl-shared/ceed-sycl-shared-basis.o
     SYCLCXX build/backends/sycl-shared/ceed-sycl-shared.o
     SYCLCXX build/backends/sycl-gen/ceed-sycl-gen-operator-build.o
     SYCLCXX build/backends/sycl-gen/ceed-sycl-gen-operator.o
     SYCLCXX build/backends/sycl-gen/ceed-sycl-gen-qfunction.o
     SYCLCXX build/backends/sycl-gen/ceed-sycl-gen.o
```
and manually checked SYCL is still fine with my update

**Comment by jeremylt** [comment]
CI is wrapping up and will pass. I think we just need an approval + merge

**Review by jedbrown — APPROVED** [review]

**Pull request merged**
